### PR TITLE
[CIAS30-3466]

### PR DESCRIPTION
**Pull request by WiktoriaRozanska**
## Related tasks
- [CIAS-3466](https://htdevelopers.atlassian.net/browse/CIAS30-3466)

## What's new?
- renamed from`verify_login_code` to `verify_account_integrity` 
- added check to verify T&C 
- return corresponding errors to these two cases (2FA and T&C)

### CODE REVIEW

**Review by msniecHTD — APPROVED** [review]

File: app/controllers/v1/auth/sessions_controller.rb [hunk 1]
```diff
@@ -1,16 +1,26 @@
...
-  def verify_login_code
+  def verify_account_integrity
+    @resource = User.find_by(email: params[:email])
+
     return unless @resource&.valid_password?(resource_params[:password])
 
-    head :forbidden if V1::Users::Verifications::Create.call(@resource, request.headers['Verification-Code'])
+    if @resource&.missing_require_fields?
+      raise ComplexException.new(I18n.t('activerecord.errors.models.user.terms_not_accepted'),
+                                 { reason: 'TERMS_NOT_ACCEPTED', require_fields: @resource.slice(:first_name, :last_name, :terms, :role) }, :forbidden)
```
Comment (WiktoriaRozanska) [inline app/controllers/v1/auth/sessions_controller.rb:18]: It should be `roles` instead of `role`

File: app/controllers/v1/users_controller.rb [hunk 2]
```diff
@@ -62,6 +62,11 @@ def confirm_logging_code
     end
   end
 
+  def confirm_terms
```
Comment (WiktoriaRozanska) [inline app/controllers/v1/users_controller.rb:65]: We need to verify somehow if this is user who has access to this account

**Review by WiktoriaRozanska — COMMENTED** [review]
we also need to fix CI

File: app/controllers/v1/users_controller.rb (hunk 2 above)
Comment (msniecHTD) [reply app/controllers/v1/users_controller.rb:65]: It is the same as in verification code. After logging in he is directly moved to this screen so in my opinion is fine to leave it as it is

Comment (WiktoriaRozanska) [reply app/controllers/v1/users_controller.rb:65]: From my perspective, however, this is a slightly different behavior. In the case of 2FA, the verification code acts as a kind of password, in addition, it is valid for 15 minutes. So if someone has it and we can be sure that he is who he claims to be. In this our case, you don't have it.

File: app/controllers/v1/auth/sessions_controller.rb (hunk 1 above)
Comment (WiktoriaRozanska) [reply app/controllers/v1/auth/sessions_controller.rb]: here, you still have a type. You will get an error here

**Pull request merged**